### PR TITLE
Gilad/popup spec

### PR DESCRIPTION
**Pull request by giladgrush**

### CODE REVIEW

File: specs/popup.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    * none - no collision detection
+  * Vertical Collision - same as horizontal
+
+## Properties
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| anchor | Element\| Node | none | Yes | The element to be used as an anchor for the popup (will open next to it). |
+| open | boolean | false | no | Whether to show or hide the Popup |
+| onOpen | Triggered when the popup is opened | null | no | Triggered when the popup is shown |
```
Comment (tobich) [inline specs/popup.md]: Events should have NOOP (() => {}) default so that they need not to be checked

File: specs/popup.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    * fit - Shifts the popup away from the edge of the viewport until it has enough space or reaches other side of viewport
+    * flipfit - Applies flip logic and then fit logic
+    * none - no collision detection
+  * Vertical Collision - same as horizontal
+
+## Properties
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| anchor | Element\| Node | none | Yes | The element to be used as an anchor for the popup (will open next to it). |
```
Comment (tobich) [inline specs/popup.md]: Suggestion: `Node | Point`. Element is an extension of Node, so you don't have to say that explicitly.

File: specs/popup.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+| open | boolean | false | no | Whether to show or hide the Popup |
+| onOpen | Triggered when the popup is opened | null | no | Triggered when the popup is shown |
+| onClose | Triggered when popup is closed | null | no | Triggered when the popup is closed |
+
+* The following props should be placed in an IPopupProps interface since they will need to be passed from higher order components.
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| anchorPosition | Object | { vertical: 'bottom', horizontal: 'left'} | no | The point on the anchor element to which the popupPosition will attach to |
```
Comment (tobich) [inline specs/popup.md]: Type `AttachementPoint` (interface)

File: specs/popup.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+| popupPosition | Object | { vertical: 'top', horizontal: 'left'} | no | The point from which the popupPosition will pivot |
```
Comment (tobich) [inline specs/popup.md]: Type `AttachementPoint` (interface)

File: specs/popup.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+| collision | Object| {'vertical: 'none', horizontal: 'none'} | No | Specify the collision behavior of the component |
```
Comment (tobich) [inline specs/popup.md]: Interface `ViewportCollision`?

File: specs/popup.md (hunk 3 above)
Comment (giladgrush) [inline specs/popup.md]: Change to IPositionPoint

File: specs/popup.md (hunk 1 above)
Comment (giladgrush) [inline specs/popup.md]: Done

File: specs/popup.md (hunk 2 above)
Comment (giladgrush) [inline specs/popup.md]: Done

File: specs/popup.md (hunk 5 above)
Comment (giladgrush) [inline specs/popup.md]: ICollision?

**Pull request merged**